### PR TITLE
Prevent FT to non-EOA account

### PR DESCRIPTION
**Pull request by JcBernack**
- prevent balance changes by forward transfers to non-EOA accounts
- add TODO that we want to return the amount to the mcReturnAddress later
- add python test to verify that a FT to an existing smart contract account will not change the balance

### CODE REVIEW

File: qa/sc_evm_forward_transfer.py [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+    SCNetworkConfiguration, LARGE_WITHDRAWAL_EPOCH_LENGTH
+from SidechainTestFramework.sc_test_framework import SidechainTestFramework
+from test_framework.util import assert_equal, assert_true, start_nodes, websocket_port_by_mc_node_index, \
+    forward_transfer_to_sidechain
+from SidechainTestFramework.scutil import bootstrap_sidechain_nodes, start_sc_nodes, \
+    is_mainchain_block_included_in_sc_block, check_mainchain_block_reference_info, AccountModelBlockVersion, \
+    EVM_APP_BINARY, generate_next_blocks, generate_next_block
+
+"""
+Check an EVM Storage Smart Contract.
```
Comment (i-Alex) [inline qa/sc_evm_forward_transfer.py:17]: Need to update the description here.

File: qa/sc_evm_forward_transfer.py [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+    - Deploy Smart Contract
+    - Execute forward transfer to the address of the smart contract
+    - Verify balance of the smart contract account has not changed
+    For the smart contract:
+        - Deploy the smart contract with initial data
+        - Check initial value
+        - Set the storage to a string
+        - Read the string in a read-only call
+        - Set the storage to a different string
+        - Read the different string in a read only call
```
Comment (i-Alex) [inline qa/sc_evm_forward_transfer.py:37]: This can be removed.

**Review by i-Alex — CHANGES_REQUESTED** [review]
Looks good. A minor change in the test description required.

**Review by alsala — APPROVED** [review]

File: qa/sc_evm_forward_transfer.py (hunk 1 above)
Comment (JcBernack) [reply qa/sc_evm_forward_transfer.py]: done

File: qa/sc_evm_forward_transfer.py (hunk 2 above)
Comment (JcBernack) [reply qa/sc_evm_forward_transfer.py]: done

**Review by i-Alex — APPROVED** [review]

**Pull request merged**